app/models/authenticator/base.rb

```diff
@@ -151,6 +151,10 @@ def authorize(taskid, username, *args)
       end
     end
 
+    def find_external_identity(_username, *_args)
+      raise NoMethodError, "required interface method 'find_external_identity' not defined by subclass"
```

Comment:
Prefer `NotImplementedError` over `NoMethodError`

Reply:
Forgot that existed. Done!